python_modules/libraries/dagster-dg/dagster_dg/cli/plus.py

```diff
@@ -63,3 +74,128 @@ def login_command() -> None:
+    envvar="DAGSTER_PLUS_ORGANIZATION",
+)
+@click.option(
+    "--deployment",
+    "deployment",
+    help="Name of the Dagster+ deployment to which to deploy. If not set, defaults to the value set by `dg plus login`.",
+    envvar="DAGSTER_PLUS_DEPLOYMENT",
+)
+@click.option(
+    "--python-version",
+    "python_version",
+    type=click.Choice(["3.9", "3.10", "3.11", "3.12"]),
+    help=(
+        "Python version used to deploy the project. If not set, defaults to the calling process's Python minor version."
+    ),
+)
+@dg_global_options
+def deploy_command(
+    organization: Optional[str],
+    deployment: Optional[str],
+    python_version: Optional[str],
+    **global_options: object,
+) -> None:
+    """Deploy a project to Dagster Plus."""
+    cli_config = normalize_cli_config(global_options, click.get_current_context())
+
+    if not python_version:
+        python_version = f"3.{sys.version_info.minor}"
+
+    plus_config = DagsterPlusCliConfig.get()
+
+    organization = organization or plus_config.organization
+    if not organization:
+        raise click.UsageError(
+            "Organization not specified. To specify an organization, use the --organization option "
+            "or run `dg plus login`."
+        )
+
+    deployment = deployment or plus_config.default_deployment
+    if not deployment:
+        raise click.UsageError(
+            "Deployment not specified. To specify a deployment, use the --deployment option "
+            "or run `dg plus login`."
+        )
+
+    # TODO This command should work in a workspace context too and apply to multiple projects
+    dg_context = DgContext.for_project_environment(Path.cwd(), cli_config)
+
+    # TODO Confirm that dagster-cloud is packaged in the project
+
+    with ExitStack() as stack:
+        # TODO Once this is split out into multiple commands, we need a default statedir
+        # that can be persisted across commands.
+        statedir = stack.enter_context(tempfile.TemporaryDirectory())
+
+        # Construct a dagster_cloud.yaml file based on info in the pyproject.toml
+        dagster_cloud_yaml_file = stack.enter_context(
+            create_temp_dagster_cloud_yaml_file(dg_context)
+        )
+
+        dg_context.external_dagster_cloud_cli_command(
+            [
+                "ci",
+                "init",
+                "--statedir",
+                str(statedir),
+                "--dagster-cloud-yaml-path",
+                dagster_cloud_yaml_file,
+                "--project-dir",
+                str(dg_context.root_path),
+                "--deployment",
+                deployment,
+                "--organization",
+                organization,
+            ],
+        )
+
+        dockerfile_path = dg_context.root_path / "Dockerfile"
+        if not os.path.exists(dockerfile_path):
+            click.echo(f"No Dockerfile found - scaffolding a default one at {dockerfile_path}.")
```

Comment:
Eventually we'd use some sort of prebuilt UV dockerfile, I'd assume, right? the equivalent of the pex base image right now

Reply:
that's more or less what the one I picked already uses i think